notifier/stomp/failover.go

```diff
@@ -0,0 +1,83 @@
+package stomp
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+
+	gostomp "github.com/go-stomp/stomp"
+	"github.com/rs/zerolog"
+)
+
+// FailOver will return the first successful connection made against the provided
+// brokers, or an existing connection if not closed.
+//
+// FailOver is safe for concurrent usage.
+type FailOver struct {
+	Config
+}
+
+// Dial will dial the provided uri in accordance with the
+// provided Config.
+func (f *FailOver) Dial(uri string) (*gostomp.Conn, error) {
+	opts := []func(*gostomp.Conn) error{}
+
+	if f.Login != nil {
+		opts = append(opts, gostomp.ConnOpt.Login(f.Login.Login, f.Login.Passcode))
+	}
+
+	var conn io.ReadWriteCloser
+	var err error
+	if f.tls != nil {
```

Comment:
In integration tests this is always nil, because `Validate` doesn't return updated config with tls setup